pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -803,25 +786,72 @@ func (dsc *ReconcileDaemonSet) podsShouldBeOnNode(
 				daemonPodsRunning = append(daemonPodsRunning, pod)
 			}
 		}
-		// If daemon pod is supposed to be running on node, but more than 1 daemon pod is running, delete the excess daemon pods.
-		// Sort the daemon pods by creation time, so the oldest is preserved.
-		if len(daemonPodsRunning) > 1 {
+
+		// When surge is not enabled, if there is more than 1 running pod on a node delete all but the oldest
+		if !allowSurge(ds) {
+			if len(daemonPodsRunning) <= 1 {
+				// There are no excess pods to be pruned, and no pods to create
+				break
+			}
+
 			sort.Sort(podByCreationTimestampAndPhase(daemonPodsRunning))
 			for i := 1; i < len(daemonPodsRunning); i++ {
 				podsToDelete = append(podsToDelete, daemonPodsRunning[i].Name)
 			}
+			break
+		}
+
+		if len(daemonPodsRunning) <= 1 {
```

Comment:
is it possible to move this if clause before allowSurge clause(Line 791), so that there is no need to check len(daemonPodsRunning) at line 793

Reply:
Maybe just let it consistent with upstream, so they can rebase more easily in future.